web/profiles/custom/os2loop/themes/os2loop_theme/os2loop_theme.theme

```diff
@@ -123,3 +126,69 @@ function os2loop_theme_theme_suggestions_container_alter(array &$suggestions, ar
     $suggestions[] = 'container' . '__' . str_replace('-', '_', $variables['element']['#id']);
   }
 }
+
+/**
+ * Implements hook_form_system_theme_settings_alter().
+ */
+function os2loop_theme_form_system_theme_settings_alter(&$form, FormStateInterface $form_state, $form_id = NULL) {
+  // Work-around for a core bug affecting admin themes. See issue #943212.
+  if (isset($form_id)) {
+    return;
+  }
+
+  $form['colors'] = [
+    '#type' => 'details',
+    '#description' => t('Select a color template or provide your own @todo link to readme. Cache must be cleared before this takes effect.'),
+    '#title' => t('Colors'),
+    '#open' => TRUE,
+  ];
+
+  $schemes = [
+    'default.css' => t('Default'),
+    'blue.css' => t('Blue'),
+    'green.css' => t('Green'),
+    'red.css' => t('Red'),
+    'yellow.css' => t('Yellow'),
+  ];
+
+  $form['colors']['color_scheme'] = [
+    '#type' => 'select',
+    '#title' => t('Select color template'),
+    '#options' => $schemes,
+    '#default_value' => theme_get_setting('color_scheme'),
+  ];
+
+  $uploaded_file = theme_get_setting('custom_scheme');
+  $form['colors']['custom_scheme'] = [
+    '#type' => 'managed_file',
+    '#name' => 'my_file',
+    '#title' => t('Custom css file'),
+    '#size' => 20,
+    '#description' => t('.css format only'),
+    '#upload_validators' => [
+      'file_validate_extensions' => ['css'],
+      'file_validate_size' => [10000000],
+    ],
+    '#default_value' => isset($uploaded_file) ? $uploaded_file : NULL,
+    '#upload_location' => 'public://custom_color_scheme/',
+  ];
+}
+
+/**
+ * Implements hook_library_info_alter().
+ */
+function os2loop_theme_library_info_alter(&$libraries, $extension) {
+  if ('os2loop_theme' === $extension) {
+    $custom_scheme = theme_get_setting('custom_scheme');
+    if (empty($custom_scheme)) {
+      $scheme = theme_get_setting('color_scheme');
+      $libraries['global-styling']['css']['theme']['build/' . $scheme] = [];
```

Comment:
Do we need `[ 'preprocess' => FALSE ]` here (cf. https://www.drupal.org/docs/theming-drupal/adding-stylesheets-css-and-javascript-js-to-a-drupal-theme#css-properties)?